src/Graph/Nodes/BinaryNode.php

```diff
@@ -16,70 +14,9 @@
 interface BinaryNode extends Node
 {
     /**
-     * Return the children of this node in a generator.
-     *
-     * @return \Traversable<\Rubix\ML\Graph\Nodes\BinaryNode>
-     */
-    public function children() : Traversable;
-
-    /**
-     * Return the left child node.
-     *
-     * @return \Rubix\ML\Graph\Nodes\BinaryNode|null
-     */
-    public function left() : ?BinaryNode;
-
-    /**
-     * Return the right child node.
-     *
-     * @return \Rubix\ML\Graph\Nodes\BinaryNode|null
-     */
-    public function right() : ?BinaryNode;
-
-    /**
-     * Recursive function to determine the height of the node in the tree.
+     * Return the height of the node in the tree.
      *
      * @return int
      */
     public function height() : int;
-
-    /**
-     * The balance factor of the node. Negative numbers indicate a
-     * lean to the left, positive to the right, and 0 is perfectly
-     * balanced.
-     *
-     * @return int
-     */
-    public function balance() : int;
-
-    /**
-     * Set the left child node.
-     *
-     * @param self $node
-     */
-    public function attachLeft(BinaryNode $node) : void;
```

Comment:
so you're dropping the attach/detach functionality. The attach is dropped from the interface but not from the trait. The detach is completely gone. I guess the detach was not being used so it sounds good. Why dropping the attach from the interface while keeping it in the trait? Just trying to understand where you want to take the design.

Reply:
Good observation. The `attach()` API was moved to the `HasBinaryChildren` interface since it is possible for some `BinaryNode`s (ex. leaf nodes) to not have children.